gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -2411,4 +2451,26 @@ private boolean requesterShouldPay(String bucketName) {
   public AtomicLong getObjectStatistics(GoogleCloudStorageStatistics key) {
     return objectStatistics.get(key);
   }
+
+  @Override
+  public AtomicLong getHttpStatistics(GoogleCloudStorageStatistics key) {
+    return gcsStatistics.get(key);
+  }
+
+  private void setHttpStatistics() {
+    if (gcsRequestsTracker != null) {
+      ImmutableList<HttpRequest> httpRequests = gcsRequestsTracker.getAllRequests();
+      httpRequests.stream()
+          .forEach(
+              req -> {
+                if (req.getRequestMethod() == "GET") {
+                  gcsStatistics.putIfAbsent(ACTION_HTTP_GET_REQUEST, new AtomicLong(1));
+                  gcsStatistics.get(ACTION_HTTP_GET_REQUEST).incrementAndGet();
+                } else if (req.getRequestMethod() == "HEAD") {
+                  gcsStatistics.putIfAbsent(ACTION_HTTP_GET_REQUEST, new AtomicLong(1));
+                  gcsStatistics.get(ACTION_HTTP_HEAD_REQUEST).incrementAndGet();
+                }
+              });
+    }
+  }
```

Comment:
If we have an open fs, and call read file twice, don't we double count the second time here ?

Reply:
Updated